src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-form/configuration-form.component.ts

```diff
@@ -0,0 +1,267 @@
+    }
+  }
+
+  getStep(type: string, value: number): number | undefined {
+    const numberTypes = ['uint64_t', 'int64_t', 'size_t', 'secs'];
+
+    if (numberTypes.includes(type)) {
+      return 1;
+    }
+
+    if (type === 'double') {
+      if (value !== null) {
+        const stringVal = value.toString();
+        if (stringVal.indexOf('.') !== -1) {
+          // Value type double and contains decimal characters
+          const decimal = value.toString().split('.');
+          return Math.pow(10, -decimal[1].length);
+        }
+      }
+
+      return 0.1;
+    }
+
+    return undefined;
+  }
+
+  private numberValidator(configOption): ValidatorFn[] {
+    const validators = [];
+    if (configOption.max && configOption.max !== '') {
+      this.maxValue = configOption.max;
+      validators.push(Validators.max(configOption.max));
+    }
+    if (configOption.min && configOption.min !== '') {
+      this.minValue = configOption.min;
+      validators.push(Validators.min(configOption.min));
+    }
+    if (configOption.type === 'double') {
+      validators.push(Validators.pattern('^-?[0-9]+(.[0-9]+)?$'));
+    } else {
+      validators.push(Validators.pattern('^-?[0-9]+$'));
+    }
+
+    return validators;
+  }
+
+  private ipv4Ipv6Validator(configOption): ValidatorFn {
+    const ipv4Ipv6Rgx =
+      '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]).){3}([0-9]|[1-9][0-9]|1[0-9]{2}|' +
+      '2[0-4][0-9]|25[0-5])$|^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9-]*[a-zA-Z0-9]).)*([A-Za-z]|[A-Za-z]' +
+      '[A-Za-z0-9-]*[A-Za-z0-9])$|^s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|((' +
+      '[0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d' +
+      '|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|' +
+      '2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}' +
+      '(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]d|1dd|[1-9]?d)' +
+      '(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4})' +
+      '{1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|' +
+      '1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|' +
+      '((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d))' +
+      '{3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:' +
+      '((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(:(((:' +
+      '[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)' +
+      '(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:)))(%.+)?s*$';
+    return Validators.pattern(ipv4Ipv6Rgx);
+  }
+
+  private uuidValidator(configOption): ValidatorFn {
+    const uuidRgx = '^[a-z0-9]{8}\\-[a-z0-9]{4}\\-[a-z0-9]{4}\\-[a-z0-9]{4}\\-[a-z0-9]{12}$';
+    return Validators.pattern(uuidRgx);
```

Comment:
What happens if the input string is empty? By default the validation should not fail in this case. Check the existing validators in CdValidators.

Reply:
Accidentally, I wrote a UUID validator, too. It will be used to validate the UUID given by the user if he decides to create an OSD with a specific UUID. As this UUID is optional, I think it does solve the problem mentioned by @votdev:

```typescript
  static uuid(required = false): ValidatorFn {
    const uuidRe = /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
    return (control: AbstractControl): { [key: string]: any } | null => {
      if (control.pristine && control.untouched) {
        return null;
      } else if (!required && !control.value) {
        return null;
      } else if (uuidRe.test(control.value)) {
        return null;
      }
      return { invalidUuid: 'This is not a valid UUID' };
    };
  }
```

Additionally, it allows the UUID to be lower- or uppercase (or mixed) as those are also valid UUIDs.

Reply:
I suggest to name it uuidv4 because there are more variation out there.

Reply:
@votdev or just fsid?

Reply:
> I suggest to name it uuidv4 because there are more variation out there.
Makes sense. I'd prefer this over `fsid` as `uuidv4` is more generic and it'd be more obvious that `uuidv4` validator can be used for OSD creation.

Reply:
Why call it fsid?

Reply:
Was just an idea.

Reply:
The validator with the regular expression provided in my example shouldn't be named `uuidv4`. It validates all different versions of UUIDs correctly.